lib/tool_shed/galaxy_install/install_manager.py

```diff
@@ -1041,10 +1035,7 @@ def fetch_tool_versions( app, tool_shed_repository ):
         if text:
             return json.loads( text )
         else:
-            log.error("No content returned from tool shed repository version request to %s", url)
-            failed_to_fetch = True
+            raise Exception("No content returned from Tool Shed repository version request to %s" % url)
     except Exception:
-        failed_to_fetch = True
-        log.exception("Failed to fetch tool shed repository version information.")
-    if failed_to_fetch:
-        return FAILED_TO_FETCH_VERSIONS
+        log.exception("Failed to fetch tool version information for Tool Shed repository.")
+        raise
```

Comment:
@nsoranzo I guess the tests breaks here because you're re-raising the exception instead of returning FAILED_TO_FETCH_VERSIONS. If you `raise` here you'd have to catch it in line 895, or only ask for the versions if you know there are tools in the repo.